project/Version.scala

```diff
@@ -0,0 +1,20 @@
+import sbt.Keys._
+import sbt._
+
+object Version {
+  val baseModelVersion = "0.1"
+  val lastModelCommit = ("git log -n 1 --pretty=format:%h model" !!).trim
+
+  val modelVersionSettings: Seq[Setting[_]] =
+    Seq( version := versionString )
+
+  val versionString = {
+    val version = baseModelVersion + "-" + lastModelCommit
+
+    // The project isSnapshot string passed in via command line settings, if desired.
+    val isSnapshot = sys.props.get("project.isSnapshot").forall(_.toBoolean)
```

Comment:
i couldn't wrap my head around this until i looked it up, so i'll save you the trouble: `forall` returns true for empty iterables.